### PR TITLE
[14.0][MIG] sale_promotion_rule_display_discount_amount

### PR DESCRIPTION
**Pull request by xavier-bouquiaux**
Depends on:

- [ ] #1583 

Remove:
- [ ] test-requirements.txt

### CODE REVIEW

File: sale_promotion_rule_display_discount_amount/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+from odoo.tools import float_compare
+
+
+class SaleOrder(models.Model):
+
+    _inherit = "sale.order"
+
+    @api.depends(
+        "order_line.discount_total",
+        "order_line.discount_total",
```
Comment (hailangvn) [inline sale_promotion_rule_display_discount_amount/models/sale_order.py]:
```suggestion
```
Is duplication necessary?

File: sale_promotion_rule_display_discount_amount/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        "order_line.is_promotion_line",
+        "order_line.currency_id",
+        "currency_id",
+    )
+    def _compute_discount_total(self):
+        super(SaleOrder, self)._compute_discount_total()
```
Comment (hailangvn) [inline sale_promotion_rule_display_discount_amount/models/sale_order.py]:
```suggestion
        super()._compute_discount_total()
```

File: sale_promotion_rule_display_discount_amount/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                        precision_rounding=order.currency_id.rounding,
+                    )
+                    != 0
+                ):
+                    order.update(
+                        {
+                            "discount_total": discount_total,
+                            "price_total_no_discount": price_total_no_discount,
+                        }
+                    )
```
Comment (hailangvn) [inline sale_promotion_rule_display_discount_amount/models/sale_order.py]:
```suggestion

            for line in order.order_line:
                if not line.is_promotion_line:
                    continue
                price_total = line.price_total
                if (
                    float_compare(
                        line.price_total,
                        0.0,
                        precision_rounding=line.currency_id.rounding,
                    )
                    < 1
                ):
                    price_total = -price_total
                discount_total += price_total
                price_total_no_discount += price_total

            if (
                float_compare(
                    discount_total,
                    order.discount_total,
                    precision_rounding=order.currency_id.rounding,
                )
                != 0
            ):
                order.update(
                    {
                        "discount_total": discount_total,
                        "price_total_no_discount": price_total_no_discount,
                    }
                )
```

File: sale_promotion_rule_display_discount_amount/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        "order_line.currency_id",
+        "currency_id",
+    )
+    def _compute_discount_total(self):
+        super(SaleOrder, self)._compute_discount_total()
+        for order in self:
+            discount_total = order.discount_total
+            price_total_no_discount = order.price_total_no_discount
+            if not order.order_line:
+                return True
```
Comment (hailangvn) [inline sale_promotion_rule_display_discount_amount/models/sale_order.py]:
```suggestion
                continue
```

File: sale_promotion_rule_display_discount_amount/tests/test_sale_order.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+from odoo.addons.sale_promotion_rule.tests.test_promotion import (
+    FIXED_AMOUNT_CODE,
+    AbstractCommonPromotionCase,
+)
+
+
+class TestSaleOrder(TransactionCase, AbstractCommonPromotionCase):
+    def setUp(self, *args, **kwargs):
+        super(TestSaleOrder, self).setUp(*args, **kwargs)
```
Comment (hailangvn) [inline sale_promotion_rule_display_discount_amount/tests/test_sale_order.py]:
```suggestion
        super().setUp(*args, **kwargs)
```

**Review by hailangvn — APPROVED** [review]
Except for minor duplicated line, all other suggestion is optional and the code is LGTM. Thanks.

**Comment by xavier-bouquiaux** [comment]
@rousseldenis , @sebastienbeau 
as we merged #1583 we could merge this ?

What do you think ?

Regards

Xbo

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**